sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -741,6 +758,144 @@ async def update_tag_properties(
+    ) -> str:
+        """Upload a manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param manifest: The manifest to upload. Note: This must be a seekable stream.
+        :type manifest: ~azure.containerregistry.models.OCIManifest or IO
+        :keyword tag: Tag of the manifest.
+        :paramtype tag: str or None
+        :returns: The digest of the uploaded manifest, calculated by the registry.
+        :rtype: str
+        :raises ValueError: If the parameter repository or manifest is None.
+        :raises ~azure.core.exceptions.HttpResponseError:
+            If the digest in the response does not match the digest of the uploaded manifest.
+        """
+        try:
+            data = manifest
+            if isinstance(manifest, OCIManifest):
+                data = _serialize_manifest(manifest)
+            tag_or_digest = tag
+            if tag is None:
+                tag_or_digest = _compute_digest(data)
+
+            _, _, response_headers = await self._client.container_registry.create_manifest(
+                name=repository,
+                reference=tag_or_digest,
+                payload=data,
+                content_type=OCI_MANIFEST_MEDIA_TYPE,
+                headers={"Accept": OCI_MANIFEST_MEDIA_TYPE},
+                cls=_return_response,
+                **kwargs
+            )
+
+            digest = response_headers['Docker-Content-Digest']
+        except ValueError:
+            if repository is None or manifest is None:
+                raise ValueError("The parameter repository and manifest cannot be None.")
+            # if not _validate_digest(data, digest):
+            #     raise ValueError("The digest in the response does not match the digest of the uploaded manifest.")
+            raise
+
+        return digest
+
+    @distributed_trace_async
+    async def upload_blob(self, repository, data, **kwargs):
+        # type: (str, IO, **Any) -> str
+        """Upload an artifact blob.
+
+        :param str repository: Name of the repository
+        :param data: The blob to upload. Note: This must be a seekable stream.
+        :type data: IO
+        :returns: The digest of the uploaded blob, calculated by the registry.
+        :rtype: str
+        :raises ValueError: If the parameter repository or data is None.
+        """
+        try:
+            _, _, start_upload_response_headers = await self._client.container_registry_blob.start_upload(
+                repository, cls=_return_response, **kwargs
+            )
+            original = data.close
+            data.close = lambda: None
+            _, _, upload_chunk_response_headers = await self._client.container_registry_blob.upload_chunk(
+                start_upload_response_headers['Location'], data, cls=_return_response, **kwargs
+            )
+            data.close = original
+            digest = _compute_digest(data)
```

Comment:
Do we have a strategy for computing the digest incrementally if the input stream represents a 10GB file stream?

Reply:
Large stream is not supported in this PR, haven't think about it yet. But I know it is in the new requirements.